wynnset/tapestry-wp
# Add Youtube support and auto-calculate video length for all videos

Closes #477 
#### Adds:
- Embedded YouTube player for video nodes (with type autodetection)
- Automatically populating `mediaDuration` field for YouTube and H5P videos
- Use H5P video settings to determine YouTube player options, and updating settings from YouTube player
____
#### Steps to test:
Basic Support:
 - Run NPM install to get YouTube component
 - Create a video node and fill in the URL section with any YouTube video link
 - After submission, verify the correct video duration appears on the node (under the title)
 - Click the media button and press play on the i-frame to begin the video

Progress:
 - Play a meaningful (> 2%) amount of the video
 - Pause the video and exit, verifying the progress bar updates 
 - Re-enter the video, and verify it begins from where you left it
 - Watch another portion, and close without pausing; verify the progress bar updates
 - Finish the video (or scroll to the end), and verify the end screen appears with the option to close, rewatch or take quiz if one is set up
 - Verify the node is marked as complete on the Tapestry

Settings:
 - Create an H5P video node
 - Open the video and alter the settings; mute, change the volume and playback speed
 - Open the YouTube video node and verify the appropriate settings have been changed. Note that playback speed changes may not work for all YouTube videos
 - Alter the settings in the YouTube player and either close or pause the video to save the changes
 - Open the H5P video node again, and verify the settings changes have saved and applied
